src/utils/event.ts

```diff
@@ -42,3 +47,30 @@ export const eventHappeningNow = (
       event.start_utc_seconds + event.duration_minutes > currentTimeInUTCSeconds
   );
 };
+
+export const eventStartTime = (event: VenueEvent) =>
+  fromUnixTime(event.start_utc_seconds);
+
+export const eventEndTime = (event: VenueEvent) =>
+  addMinutes(eventStartTime(event), event.duration_minutes);
+
+export const isEventStartingSoon = (event: VenueEvent) =>
+  differenceInHours(eventStartTime(event), Date.now()) > 0;
+
+export const getEventInterval = (event: VenueEvent) => ({
+  start: eventStartTime(event),
+  end: eventEndTime(event),
+});
+
+export const isEventWithinDate = (checkDate: number | Date) => (
+  event: VenueEvent
+) => {
+  const checkDateInterval = {
+    start: startOfDay(checkDate),
+    end: endOfDay(checkDate),
+  };
+
+  const eventInterval = getEventInterval(event);
+
+  return areIntervalsOverlapping(checkDateInterval, eventInterval);
```

Comment:
[NB] If you extract `getDayInterval` into a helper function as suggested above, + remove the `{`, `}`, `return`, then we could make this a 1 liner:


```suggestion
  areIntervalsOverlapping(getDayInterval(date), getEventInterval(event));
```

Reply:
simplified: https://github.com/sparkletown/sparkle/pull/1426/commits/64e2b9d2c1f4b2cf715ff86330b85947d012a966